pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -211,8 +238,12 @@ func (rc *reconciler) reconcile() {
 
 			// Trigger detach volume which requires verifying safe to detach step
 			// If timeout is true, skip verifySafeToDetach check
+			// If the node has node.kubernetes.io/out-of-service taint with NoExecute effect, skip verifySafeToDetach check
 			klog.V(5).InfoS("Starting attacherDetacher.DetachVolume", "volume", attachedVolume)
-			verifySafeToDetach := !timeout
+			if hasOutOfServiceTaint {
+				klog.V(5).Infof("node %q has out-of-service taint", attachedVolume.NodeName)
```

Comment:
I think we can use level 4 log here.

Reply:
Sure. Changed it. 
Also, trying to learn here: How do we decide on when to use what log level when we have around 1-10? -- Like I tried to find some pattern in the existing code base but could not grok much.

Reply:
Just my typical use of logging level. For very useful and critical log, set it to level 2 or below. For useful log for debugging purpose, we can set it level 4. I rarely use 5 or above since we typically do not set logging to that high.

Reply:
Thank you. 🙏  This helps.